core/trino-main/src/main/java/io/trino/execution/executor/timesharing/MultilevelSplitQueue.java

```diff
@@ -158,20 +159,17 @@ public PrioritizedSplitRunner take()
     @GuardedBy("lock")
     private PrioritizedSplitRunner pollSplit()
     {
-        long targetScheduledTime = getLevel0TargetTime();
-        double worstRatio = 1;
+        double minScheduledTimeWithoutWeight = Double.MAX_VALUE;
         int selectedLevel = -1;
         for (int level = 0; level < LEVEL_THRESHOLD_SECONDS.length; level++) {
             if (!levelWaitingSplits[level].isEmpty()) {
                 long levelTime = levelScheduledTime[level].get();
-                double ratio = levelTime == 0 ? 0 : targetScheduledTime / (1.0 * levelTime);
-                if (selectedLevel == -1 || ratio > worstRatio) {
-                    worstRatio = ratio;
+                double scheduledTimeWithoutWeight = levelTime == 0 ? 0 : levelTime / levelWeight[level];
```

Comment:
Check if `levelTime` is zero, is not needed anymore. We can check if `levelWeight[level]` is zero (but we can skip it).

Reply:
agree, tks